pkg/velero/server/config.go

```diff
@@ -81,8 +81,8 @@ type ServerConfig struct {
 	// Backup all volumes with pod volume file system backup by default.
 	// +optional
 	DefaultVolumesToFsBackup *bool `json:"default-volumes-to-fs-backup,omitempty"`
-	// uploaderType, oadp only support restic at the moment
-
+	// The type of uploader to transfer the data of pod volumes, the supported values are 'restic', 'kopia' (default "restic")
```

Comment:
here can't we add validation enum and remove code validation from `pkg/velero/server/args.go`?

Reply:
I still prefer to have validation on the `pkg/velero/server/args.go` side.

The earlier the user is informed the fields are incorrect the better, which makes that is why we have for that field enumeration that only allows two entries:

```
	// +kubebuilder:validation:Enum=restic;kopia
```

However it's a code side responsibility to confirm that arguments used in the functions are correct. From the go language I don't have the warranty that the function ` StringArr` will *always* receive only those two fields, that's why I do prefer to have such validation in that function.

The `server/args.go` is mostly used outside of standard OADP usage (development/testing) and as such should show also errors when one would be using that function in non-standard way.

Reply:
Commented out that part, so it's not relevant anymore.  See link on the commented out part.